### DISCUSSION TITLE
Positive Only Ticket Karma

### DISCUSSION THREAD

**Discussion by Etron-RA** [RetroAchievements/RAWeb#1010] [Ideas]
I’ve been thinking about a solution to the growing ticket crisis and the negative stigma surrounded by tickets.  Ideally we want developers to be motivated to resolve/close tickets for inactive developers and their own tickets.  I believe the original intent of ticket karma was to reward developers for solving others’ tickets however, the negative karma points for receiving a ticket makes the getting a ticket feel like a punishment.  This is compounded by players that don’t fully understand the ticket system and use the tickets for achievement suggestions or to report 70 tickets for a BIOS issue on their end.

To remedy this I purposed we revamp the ticket karma system to be only positive points.  We expect developers to resolve their own tickets however, it can be a lot of work to figure out a solution and we should reward the developer for addressing issues with their sets.  To promote resolving tickets for inactive developers the karma reward should be greater since it is above and beyond the call of duty.  Ultimately we want to improve the quality of sets on the site so the focus should be on closing or resolving tickets.

The initial plan is to reward points based on the achievement owner:
1 pts = You’re the owner
2 pts = Another active dev (assuming permission is given or ticket is 30+ days old)
4 pts = Registered user
8 pts = Banned user

I believe that this plan could be implemented retroactively since we have the ticket owner and resolver for every ticket in the database.  The karma sum should be dynamic as the players status will change overtime.  The result would be everyone will get a positive ticket karma and developers that resolve others' tickets will have a higher karma rating.  As before, anyone caught abusing the ticket system would get a warning/reprimand from Devcompliance.  Tickets are a fact of dev life, let’s end the negative ticket stigma!

**Comment by TeddyWestside22** [comment]
Receiving a negative increase despite resolving a ticket is the biggest issue with the current system and makes it pointless. If you have an open ticket, sure, that can be counted as negative, but once it is resolved/closed, it should return to 0. 

As a karma system, there should certainly be incentives to help resolve other (inactive) developer tickets. I'm on board with the +1 for your own tickets (which would return it to 0), but I'm not sure it makes a difference in the other cases whether they're a developer, registered user or banned user. I think +1 to all would be fine.

Here's a mockup of how it could be done: 

![image](https://user-images.githubusercontent.com/32810074/183895394-57b859a0-cf86-4f89-8636-1a09ca660c07.png)

It could then be displayed on your profile, which would provide further motivation to continue to help others:

![image](https://user-images.githubusercontent.com/32810074/183895530-e36963dc-aed9-4a2e-970b-bfce0dae6e2e.png)

The calculation is simplified, but the current one doesn't really tell us anything of significance. It's how many tickets resolved/closed for others divided by how many tickets you have received. Given that tickets can be made for random things like the mentioned BIOS issues or others completely unrelated to the logic, it's not useful information.

**Reply by Tsearo** [reply]
This is personally how I would like to see it handled, just keeping it simple and changing it to be how many tickets you have resolved or closed by other developers like you've shown.  Though tbf I really don't mind how it is handled in the end as long as it becomes a way to track positive handling of tickets.

Showing it on the user page like that would be a nice addition as well.

One other thing I've thought would be nice would be to add it to the Developer Stats page with a sortable column to show devs that have contributed the most towards working on others tickets.

**Reply by redwizard42** [reply]
Yeah, that's also how I'd want it to work. With the slight grumble desire for 'mistaken report' to add back 2 karma :D

**Comment by Jamiras** [comment]
#1203 eliminates the karma system. Instead a developer's "rating" is determined by how many of their achievements have been ticketed. There is no penalty if someone other than the author resolves the ticket.

"Tickets Resolved for Others" is still being tracked, with the option to add tiered badges for it in the future. It does not contribute to the developer's rating.
